pkg/rtc/room.go

```diff
@@ -445,36 +457,36 @@ func (r *Room) UpdateSubscriptions(
 	subscribe bool,
 ) error {
 	// find all matching tracks
-	trackPublishers := make(map[livekit.TrackID]types.Participant)
+	publisherIDs := make(map[livekit.TrackID]livekit.ParticipantID)
+	publisherIdentities := make(map[livekit.TrackID]livekit.ParticipantIdentity)
 	participants := r.GetParticipants()
 	for _, trackID := range trackIDs {
 		for _, p := range participants {
 			track := p.GetPublishedTrack(trackID)
 			if track != nil {
-				trackPublishers[trackID] = p
+				publisherIDs[trackID] = p.ID()
+				publisherIdentities[trackID] = p.Identity()
 				break
 			}
 		}
 	}
 
 	for _, pt := range participantTracks {
-		p := r.GetParticipantBySid(livekit.ParticipantID(pt.ParticipantSid))
-		if p == nil {
-			continue
-		}
+		pub := r.GetParticipantByID(livekit.ParticipantID(pt.ParticipantSid))
 		for _, trackID := range livekit.StringsAsTrackIDs(pt.TrackSids) {
-			trackPublishers[trackID] = p
+			publisherIDs[trackID] = livekit.ParticipantID(pt.ParticipantSid)
+			if pub != nil {
+				publisherIdentities[trackID] = pub.Identity()
+			}
 		}
 	}
 
 	// handle subscription changes
-	for trackID, publisher := range trackPublishers {
+	for trackID, publisherID := range publisherIDs {
 		if subscribe {
-			if _, err := publisher.AddSubscriber(participant, types.AddSubscriberParams{TrackIDs: []livekit.TrackID{trackID}}); err != nil {
-				return err
-			}
+			participant.SubscribeToTrack(trackID, publisherIdentities[trackID], publisherID)
```

Comment:
Can't say with certainty as it is hard to keep track in my head :-), but `SubscribeToTrack` may not be handling `nil` identity? It can be `nil` here. Is that handled properly? Or maybe this place should not call `SubscribeToTrack` if `identity` is `nil`?

Reply:
yeah it's possible for participantIdentity to be empty. We do not rely on publisher identity in the subscription stack, so this is just for improved logging.